src/hip_hcc.cpp

```diff
@@ -1082,6 +1094,14 @@ void ihipCtx_t::locked_syncDefaultStream(bool waitOnSelf, bool syncHost) {
             defaultCf.wait();  // TODO - account for active or blocking here.
         }
     }
+    else if ( (HIP_SYNC_NULL_STREAM && !last_stream_waited) ||
+              (!HIP_SYNC_NULL_STREAM && depOps.empty()) ) {
+        // This catches all the conditions where the printf buffer
+        // need to be explicitly flushed
+        if (syncHost) {
```

Comment:
Should `syncHost` check be the first order check instead of nested inside? Could make the code simpler to read.

Reply:
Same comment as above.  Please open a new PR to address this.

Reply:
Keeping `syncHost` nested here is preferred since it follows the same structure as in the previous `if` branch

Reply:
@scchan While consistency is a good thing, in this case, it seems might be more efficient to test `syncHost` or `withThisStream` first. and could also make the code more readable.